.github/workflows/CI_CD.yml

```diff
@@ -0,0 +1,84 @@
+name: CI/CD
+
+on:
+  push:
+    branches: [ main, dev ]
+  pull_request:
+    branches: [ main, dev ]
+
+jobs:
+  CI:
+    runs-on: ubuntu-latest
+
+    services:
+      db:
+        image: postgres:11
```

Comment:
Are we using pg? I would argue that we'll stick to SQLite3 for both *dev* and *prod*, for the sake of simplicity. I don't foresee this app needing the power of Postgres anytime soon.

I removed pg from the Gemfile... If pg is to be used, it would need to be added there as well: https://github.com/hpi-swt2/rolodex-portal/blob/preparation/Gemfile

Reply:
I did not realize, you removed Postgres. Then, of course, it does not make sense to use that for the CI-pipeline.

Reply:
Services in GitHub Actions are run in Docker containers. The "image" key defines, what kind of image is supposed to be loaded to the docker container. That can either be an image name from the Docker Hub or a registry name (whatever is meant by the latter) [https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idservices](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idservices). 

Postgres published an official docker image that can be used as how I did it in the workflow file. Unfortunately, I could not find anyone using SQLite3 in a GitHub Action and there is no official SQLite3 image on Docker Hub, just some community images. 

Because of that, I'm not sure how to apply SQLite3 to a GitHub Action (this is only the second time, I try to configure a GitHub Action).

So what do we want to do? Use pg because it is easier to put in the workflow, use a community version of SQLite3, write an own image file, or do something I haven't thought of yet?

Reply:
Thanks for investigating. Quite a nice feature of GH Actions, that external services (such as pg in our case) run in a separate Docker container!
[The documentation](https://docs.github.com/en/free-pro-team@latest/actions/guides/about-service-containers) says:
> Service containers are Docker containers that provide a simple and portable way for you to host services that you might need to test or operate your application in a workflow.

My argument would be that due to [the nature of SQLite](https://en.wikipedia.org/wiki/SQLite), no separate service/Docker image is necessary and that SQLite can be installed like any other gem dependency. This would mean getting rid of the `services` part of the yaml and letting the `bundle install` command handle everything (possibly prededed by a `sudo apt install libsqlite3-dev`, if that's not already available). Would you give that a shot?

Reply:
done